src/include/sof/ipc/common.h

```diff
@@ -90,6 +90,13 @@ struct ipc {
 #define ipc_get_drvdata(ipc) \
 	((ipc)->private)
 
+struct ipc_core_ctx {
+	int error;
```

Comment:
this may not always be an error, but its more of a reply with context. maybe call it ipc4_reply.